pkg/sync/syncer_test.go

```diff
@@ -81,6 +81,107 @@ func TestExpandGrants(t *testing.T) {
 	_ = os.Remove(c1zpath)
 }
 
+func TestExpandGrantImmutable(t *testing.T) {
```

Comment:
is it possible to add a test for a circular group expansion?  i think this current test is a fine validation, but i was wondering about the behavoir of full circles, eg group A has member B which has member group C which has member A -> and a user could be attached at any point, but only have direct in a single group?

Reply:
I added a test but it looks like we'll need more changes to support immutable in circular group membership.